### PR TITLE
Sass 2180 - Create 'UK pension income' task list (summary list page)

### PR DESCRIPTION
**Pull request by HarryGull**
income from pensions - Create 'UK pension income' task list (summary list page) 

Add a link to the relevant story in Jira
[SASS-2180](https://jira.tools.tax.service.gov.uk/browse/SASS-2180)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/controllers/pensions/incomeFromPensions/StatePensionController.scala [hunk 1]
```diff
@@ -51,8 +51,10 @@ class StatePensionController @Inject()(implicit val cc: MessagesControllerCompon
 
   def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
     inYearAction.notInYear(taxYear) {
+
       pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
         case Some(data) =>
+          val hh: Boolean =data.pensions.incomeFromPensions.statePension.flatMap(_.amountPaidQuestion).contains(true)
```
Comment (ILee15) [inline app/controllers/pensions/incomeFromPensions/StatePensionController.scala:57]: Is this line meant to be here?

File: it/controllers/pensions/incomeFromPensions/UkPensionIncomeSummaryControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(expectedTitle)
+          h1Check(expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(pensionName1, pensionNameSelector(1))
+          textOnPageCheck(pensionName2, pensionNameSelector(2))
+          
+          //TODO: replace hrefs "#" below with link to first details page when available .e.g. PensionSchemeDetailsController.show(taxYearEOY, Some(1)).url
```
Comment (ILee15) [inline it/controllers/pensions/incomeFromPensions/UkPensionIncomeSummaryControllerISpec.scala:119]: I think the change link has been changed to redirect to the pension specific CYA (slide 10 on the design slides), could you update the todos?

File: app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+@pensionIncomeSummaryList = {
+    @if(pensionIncomeList.nonEmpty){
+        <div class="govuk-form-group" id="pensionIncomeSummaryList">
+            <dl class="hmrc-add-to-a-list hmrc-add-to-a-list--short">
+                @for((pensionIncome, index) <- pensionIncomeList.zipWithIndex){
+                    <div class="hmrc-add-to-a-list__contents">
+                        <dt class="hmrc-add-to-a-list__identifier hmrc-add-to-a-list__identifier--light">
+                            @pensionIncome.pensionSchemeName.getOrElse("")
```
Comment (ILee15) [inline app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html:56]: Should we allow it to show a blank space? I think the pension scheme name is required in the API and they would have entered it in the flow before it gets added to this list so the Scheme name should always exist. Not sure how to handle this, could be changing the model or doing it elsewhere?

File: app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                        </dt>
+                        <dd class="hmrc-add-to-a-list__change">
+                        @*TODO:Update the link to the pension details page with index when page available like below:
+                            i.e. replace "#"with "@UkPensionSchemeDetailsController.show(taxYear, Some(index))"
```
Comment (ILee15) [inline app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html:60]: the change link has been changed to redirect to the pension specific CYA (slide 10 on the design slides), could you update the todos?

File: app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html (hunk 3 above)
Comment (HarryGull) [reply app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html:56]: It's not a space but just empty
this is just defensive coding - shouldn't happen in practice but cannot display a name we don't have
Should it encounter this it just won't display it and the user can change it
Edge case but seems fine to me

File: it/controllers/pensions/incomeFromPensions/UkPensionIncomeSummaryControllerISpec.scala (hunk 2 above)
Comment (HarryGull) [reply it/controllers/pensions/incomeFromPensions/UkPensionIncomeSummaryControllerISpec.scala]: ok

File: app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html (hunk 3 above)
Comment (HarryGull) [reply app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html:56]: It's not a space but just empty
this is just defensive coding - shouldn't happen in practice but cannot display a name we don't have
Should it encounter this it just won't display it and the user can change it
Edge case but seems fine to me

File: app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html (hunk 4 above)
Comment (HarryGull) [reply app/views/pensions/incomeFromPensions/UkPensionIncomeSummary.scala.html]: ok

**Review by ILee15 — APPROVED** [review]

**Pull request merged**